specification/vmwarevirtustream/resource-manager/Microsoft.VMwareVirtustream/preview/2016-09-01-preview/vmwarevirtustream.json

```diff
@@ -0,0 +1,1199 @@
+        },
+        "authorizations": {
+          "type": "array",
+          "description": "Authorizations for the ExpressRoute (Microsoft Colo only)",
+          "items": {
+            "$ref": "#/definitions/ExpressRouteAuthorization"
+          }
+        }
+      }
+    },
+    "ClusterDetail": {
+      "properties": {
+        "clusterSize": {
+          "type": "integer",
+          "description": "Number of hosts in the cluster"
+        }
+      }
+    },
+    "Vpc": {
+      "type": "string"
+    },
+    "Endpoints": {
+      "properties": {
+        "nsxManager": {
+          "type": "string"
+        },
+        "vcsa": {
+          "type": "string"
+        }
+      }
+    },
+    "AddIdentitySourceRequest": {
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "alias": {
+          "type": "string"
+        },
+        "domain": {
+          "type": "string"
+        },
+        "baseUserDN": {
+          "type": "string"
+        },
+        "baseGroupDN": {
+          "type": "string"
+        },
+        "primaryServer": {
+          "type": "string"
+        },
+        "secondaryServer": {
+          "type": "string"
+        },
+        "useSsl": {
+          "type": "boolean"
+        },
+        "username": {
+          "type": "string"
+        },
+        "credential": {
+          "type": "string"
+        }
+      }
+    },
+    "DeleteIdentitySourceRequest": {
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "alias": {
+          "type": "string"
+        },
+        "domain": {
+          "type": "string"
+        }
+      }
+    },
+    "IdentitySource": {
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "alias": {
+          "type": "string"
+        },
+        "domain": {
+          "type": "string"
+        },
+        "baseUserDN": {
+          "type": "string"
+        },
+        "baseGroupDN": {
+          "type": "string"
+        },
+        "primaryServer": {
+          "type": "string"
+        },
+        "secondaryServer": {
+          "type": "string"
+        },
+        "useSsl": {
+          "type": "boolean"
+        }
+      }
+    },
+    "PrivateCloud": {
+      "properties": {
+        "provisioningState": {
+          "type": "string",
+          "enum": [
+            "Succeeded",
+            "Failed",
+            "Cancelled",
+            "Pending",
+            "Building",
+            "Updating"
+          ]
+        },
+        "circuit": {
+          "$ref": "#/definitions/Circuit"
+        },
+        "cluster": {
+          "$ref": "#/definitions/Cluster"
+        },
+        "clusters": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "endpoints": {
+          "$ref": "#/definitions/Endpoints"
+        },
+        "internetEnabled": {
```

Comment:
typically bools are not recommended. they are less descriptive and dont allow for future expansion of states/values the property can take. Is it possible to model this as an enum?

Reply:
I moved these all to enums as requested for the next API version.